vip-application/src/main/java/fr/insalyon/creatis/vip/application/client/bean/boutiquesTools/BoutiquesApplication.java

```diff
@@ -220,6 +222,40 @@ public String getJsonLFN() {
         return this.applicationLFN + "/json/" + getName() + ".json";
     }
 
+    public String getCustomProperties() {
+        if (customProperties == null || customProperties.isEmpty()) {
+            return "";
+          }
+          // Use a StringBuilder to efficiently build the string
+          StringBuilder formattedString = new StringBuilder();
+          boolean first = true;
+          for (String property : customProperties) {
+            if (!first) {
+              formattedString.append(", "); // Add comma and space after first element
+            }
+            formattedString.append(property);
+            first = false;
+          }
+          return formattedString.toString();
+    }
+
+    public String getCommandLineFlag() {
+        if (commandLineFlag == null || commandLineFlag.isEmpty()) {
+            return "";
+          }
+          // Use a StringBuilder to efficiently build the string
+          StringBuilder formattedString = new StringBuilder();
+          boolean first = true;
+          for (String property : commandLineFlag) {
+            if (!first) {
+              formattedString.append(", ");
+            }
+            formattedString.append(property);
+            first = false;
+          }
+          return formattedString.toString();
+    }
+
```

Comment:
This class must remain a simple data storage class with only getter/setter and almost any inteligence.
`getCustomProperties` must return the raw `Set<String>` of the field.
`getCommandLineFlag` must be done elsewhere.

Reply:
moved all intelligence, and made this a simple getter setter class